### PR TITLE
APIS-6625 - Better errors from publisher

### PR DESCRIPTION
**Pull request by AndySpaven**

### CODE REVIEW

File: it/uk/gov/hmrc/apipublisher/PublisherFeatureSpec.scala [hunk 1]
```diff
@@ -117,13 +117,41 @@ class PublisherFeatureSpec extends BaseFeatureSpec {
...
+      Then("The api-publisher responded with status BAD_REQUEST")
+      publishResponse.code shouldBe BAD_REQUEST // TODO
+
+      And("The validation errors are present in the response body")
+      val responseBody: JsValue = Json.parse(publishResponse.body)
+      responseBody shouldBe Json.obj(
+        "code"    -> JsString(ErrorCode.INVALID_API_DEFINITION.toString),
+        "message" -> JsString("Unable to find definition for service test.example.com")
+      )
+      println("XXXXXX")
```
Comment (johnsgp) [inline it/uk/gov/hmrc/apipublisher/PublisherFeatureSpec.scala:151]: println to remove

File: it/uk/gov/hmrc/apipublisher/PublisherFeatureSpec.scala [hunk 2]
```diff
@@ -117,13 +117,41 @@ class PublisherFeatureSpec extends BaseFeatureSpec {
 
       And("The validation errors are present in the response body")
       val responseBody: JsValue      = Json.parse(publishResponse.body)
-      (responseBody \ "code").as[String] shouldBe INVALID_API_DEFINITION.toString
+      (responseBody \ "code").as[String] shouldBe ErrorCode.INVALID_API_DEFINITION.toString
+      println("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
+      println(responseBody)
```
Comment (johnsgp) [inline it/uk/gov/hmrc/apipublisher/PublisherFeatureSpec.scala:122]: printlns to remove

File: it/uk/gov/hmrc/apipublisher/PublisherFeatureSpec.scala [hunk 3]
```diff
@@ -117,13 +117,41 @@ class PublisherFeatureSpec extends BaseFeatureSpec {
...
+
+      When("The publisher is triggered")
+      val publishResponse: HttpResponse[String] =
+        Http(s"$serverUrl/publish")
+          .header(CONTENT_TYPE, JSON)
+          .header(AUTHORIZATION, encodedPublishingKey)
+          .postData(s"""{"serviceName":"test.example.com", "serviceUrl": "$apiProducerUrl", "metadata": { "third-party-api" : "true" } }""").asString
+
+      Then("The api-publisher responded with status BAD_REQUEST")
+      publishResponse.code shouldBe BAD_REQUEST // TODO
```
Comment (johnsgp) [inline it/uk/gov/hmrc/apipublisher/PublisherFeatureSpec.scala:143]: Is this still TODO?

**Comment by johnsgp** [comment]
Does README.md need to be updated?

**Review by mattclark-zerogravit — APPROVED** [review]
👍

**Comment by AndySpaven** [comment]
Done thru squash into another branch with signing fixed

**Pull request closed without merging**